lib/scraper.js

```diff
@@ -99,6 +99,10 @@ Scraper.prototype.getResourceHandler = function getHandler (resource) {
 Scraper.prototype.loadResource = function loadResource (resource) {
 	var self = this;
 
+	if(!self.options.urlFilter(resource.url)){
+		return Promise.resolve(null);
```

Comment:
Yep, I agree it needs huge refactoring
Code was written long time ago without any architecture as quick script to solve my needs and then migrated to module
But it works and people use it. I'm afraid of doing huge refactor now